### ISSUE TITLE
[NBug] Operations that change non-concurrent collections must have exclusive access.

### ISSUE THREAD

**Issue by mstv** [gitextensions/gitextensions#11460]
### Environment

- Git Extensions 33.33.33
- Build c4a15c30bae0f972e505f1121ed1813305c2d237
- Git 2.42.0.windows.1
- Microsoft Windows NT 10.0.19045.0
- .NET 8.0.0
- DPI 96dpi (no scaling)
- Portable: False

### Issue description

```
System.InvalidOperationException: Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct.
   at void GitUI.RevisionGridControl.PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs, bool forceRefreshRefs)+() => { } in D:/Build/gitextensions3/GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:line 1352
   at async Task GitUI.TaskManager.HandleExceptionsAsync(Func<Task> asyncAction, Func<Exception, Task> handleExceptionAsync) in D:/Build/gitextensions3/GitExtUtils/GitUI/TaskManager.cs:line 28
```

### Steps to reproduce

try to open FormFileHistory for `FormBrowse.cs`

### Did this work in previous version of GitExtensions?

yes

**Comment by sharwell** [comment]
Here's the relevant code:
https://github.com/gitextensions/gitextensions/blob/c4a15c30bae0f972e505f1121ed1813305c2d237/GitUI/UserControls/RevisionGrid/RevisionGridControl.cs#L1351-L1352

The true source of the exception is getting lost because the `throw exception` here is overwriting the original stack trace. It should instead do this:

```
ExceptionDispatchInfo.Throw(exception);
```

**Comment by mstv** [comment]
Thank you, @sharwell!

**Comment by sharwell** [comment]
Note that the change I suggested won't fix the underlying bug. It will simply make it so the next time it occurs, it will have a more usable stack trace for investigation.

**Comment by mstv** [comment]
```
System.InvalidOperationException: Operations that change non-concurrent collections must have exclusive access. A concurrent update was performed on this collection and corrupted its state. The collection's state is no longer correct.
   at bool System.Collections.Generic.Dictionary<TKey, TValue>.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at string GitUI.RevisionGridControl+<>c__DisplayClass145_0.<PerformRefreshRevisions>g__BuildPathFilter|5(?)+BuildPathFilter(string path) in D:/Build/gitextensions3/GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:line 1222
   at void GitUI.RevisionGridControl.PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs, bool forceRefreshRefs)+() => { } in D:/Build/gitextensions3/GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:line 1079
   at void GitUI.TaskManager.HandleExceptions(Action action, Action<Exception> handleException) in D:/Build/gitextensions3/GitExtUtils/GitUI/TaskManager.cs:line 47
   at void GitUI.RevisionGridControl.PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs, bool forceRefreshRefs)+() => { } in D:/Build/gitextensions3/GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:line 1357
   at Func<Task> GitUI.TaskManager.AsyncAction(Action action)+() => { } in D:/Build/gitextensions3/GitExtUtils/GitUI/TaskManager.cs:line 63
   at async Task GitUI.TaskManager.HandleExceptionsAsync(Func<Task> asyncAction, Func<Exception, Task> handleExceptionAsync) in D:/Build/gitextensions3/GitExtUtils/GitUI/TaskManager.cs:line 28
```

occurs with #11205 only
avoided by 2cb7fffad147c7b2afe6b3864ab12cf3c12acf09
double refresh resolved by 0272c286b17ecbe2ae250ec7e87d1e67a6ac0a72
